sdk/eventgrid/Microsoft.Azure.WebJobs.Extensions.EventGrid/tests/TestListener.cs

```diff
@@ -127,6 +132,150 @@ public async Task TestCloudEvent()
             Assert.AreEqual(HttpStatusCode.Accepted, response.StatusCode);
         }
 
+        [Test]
+        public async Task TestSingleCloudEventWithTracing()
+        {
+            // individual elements
+            var ext = new EventGridExtensionConfigProvider(new HttpRequestProcessor(NullLoggerFactory.Instance.CreateLogger<HttpRequestProcessor>()), NullLoggerFactory.Instance);
+            var host = TestHelpers.NewHost<MyProg1>(ext);
+            await host.StartAsync(); // add listener
+
+            using var testListener = new ClientDiagnosticListener("Azure.Messaging.EventGrid");
+            const string functionName = "TestEventGrid";
+            const string traceparent = "00-0123456789abcdef0123456789abcdef-0123456789abcdef-01";
+            var request = CreateSingleRequest(functionName,
+                JObject.Parse($"{{'subject':'one','data':{{'prop':'alpha'}}, 'traceparent':'{traceparent}'}}"));
```

Comment:
Can we have at least one test use an actual CloudEvent and serialize using the built-in System.Text.Json converter?

https://github.com/Azure/azure-sdk-for-net/blob/4162f6fa2445b2127468b9cfd080f01c9da88eba/sdk/core/Azure.Core/src/Messaging/CloudEventConverter.cs